### PR TITLE
Fixing the data type of maxSpaceForReqs and review of primaryCombinedReqSize

### PR DESCRIPTION
**Pull request by arc-vmware**
Following 2 issues are handled in this PR
1) The datatype of maxSpaceForReqs is uint_16. This variable
stores the size of max space needed for the PrePrepareMsg.
So this variable can limit the size of PPM to 64KB.
2) The primaryCombinedReqSize variable should keep track of
actual total size of ClientRequestMsg. So every time we push CRM
into PPM, we have to increment it. And every time we pop CRM out of PPM
we have to reduce it with the removed CRM size. In this PR we will
review and change if we see some issues.

### CODE REVIEW

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 1]
```diff
@@ -269,8 +269,8 @@ void ReplicaImp::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
                                                         << "], senderId=" << senderId);
         if (time_to_collect_batch_ == MinTime) time_to_collect_batch_ = getMonotonicTime();
         requestsQueueOfPrimary.push(m);
-        primary_queue_size_.Get().Set(requestsQueueOfPrimary.size());
         primaryCombinedReqSize += m->size();
+        primary_queue_size_.Get().Set(requestsQueueOfPrimary.size());
```
Comment (yuliasherman) [inline bftengine/src/bftengine/ReplicaImp.cpp:273]: Why did you change the order of operations?

**Review by yuliasherman — APPROVED** [review]
LGTM

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 1 above)
Comment (arc-vmware) [reply bftengine/src/bftengine/ReplicaImp.cpp:273]: I am trying to follow the following conventions:
1) Whenever we push into requestsQueueOfPrimary, we update the (byte)size in  primaryCombinedReqSize.
2) Whenever we pop from the requestsQueueOfPrimary we do the following:
    a) Access the front() of the queue in some variable say 'f' (its a shared ptr to CRM).
    b) Update the (byte)size in  primaryCombinedReqSize.
    c) Pop the queue.
    d) delete the shared ptr.
